File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = new Map();

    for (const [ index, num ] of nums.entries()) {
        if (map.has(target - num)) {
            return [ map.get(target - num), index]
        } else {
            map.set(num, index);
        }
    }
    return [];
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        obj = {}
        for i, num in enumerate(nums):
            if target - num in obj:
                return [ obj[target-num], i ]
            else:
                obj[num] = i
        return []
        